mteb/abstasks/AbsTaskMultilabelClassification.py

```diff
@@ -36,6 +38,20 @@ def evaluate_classifier(
     return scores
 
 
+def _multihot_encoder(labels):
+    """Convert list of label lists into a 2-D multihot numpy array"""
+    label_set = set()
+    for label_list in labels:
+        label_set = label_set.union(set(label_list))
+    label_set = sorted(label_set)
+
+    multihot_vectors = []
+    for label_list in labels:
+        multihot_vectors.append([1 if x in label_list else 0 for x in label_set])
+
+    return np.array(multihot_vectors)
+
+
 class AbsTaskMultilabelClassification(AbsTask):
```

Comment:
The problem is that this behaviour is not only needed in tasks that always have multiple labels. I implemented this behaviour in AbsTask originally as a backoff for when single label stratification fails, I think we should probably go for that approach.

Reply:
I don't quite get it why we should even try single label stratification on a multilabel problem?

Reply:
Cool, thanks for elaborating. I see your points. If useful in other tasks, it should definately be in AbsTask. Though, the place I don't like are those conditions:

```
            try:
                dataset_dict = dataset_dict.class_encode_column(label)
            except ValueError as e:
                if isinstance(dataset_dict[splits[0]][label][0], Sequence):
                    ...
```
IMHO we should just recognise by task type and apply stratified split where it really makes sense, instead of relying on try/except.

Reply:
We can't do it, cause ClusteringFast is both hierarchical and flat unfortunately.